base/strings/basic.jl

```diff
@@ -135,17 +135,44 @@ end
 
 ## Generic indexing functions ##
 
-prevind(s::DirectIndexString, i::Integer) = i-1
-prevind(s::AbstractArray   , i::Integer) = i-1
-nextind(s::DirectIndexString, i::Integer) = i+1
-nextind(s::AbstractArray   , i::Integer) = i+1
+prevind(s::DirectIndexString, i::Integer) = convert(Int, i)-1
+prevind(s::AbstractArray   , i::Integer) = convert(Int, i)-1
```

Comment:
`Int(i-1)` would be a little more concise - the alignment also looks funny here, though it was that way before your change